lib/src/main/java/uk/gov/di/ipv/core/library/service/AccessTokenService.java

```diff
@@ -64,4 +73,23 @@ public void persistAccessToken(AccessTokenResponse tokenResponse, String ipvSess
         accessTokenItem.setIpvSessionId(ipvSessionId);
         dataStore.create(accessTokenItem);
     }
+
+    public ValidationResult<ErrorObject> extractJwt(String requestBody) {
```

Comment:
It might be nicer to move this kind of thing into a Validator class, similar to the AuthRequestValidator that we have now?

Reply:
Also might be worth renaming this method, it's starting to do a bit more then just extract the JWT from the query params. Especially if the extra validation needed in ticket PYI-684 is going into here as well

Reply:
move extractJwt method to AuthRequestValidator class

Reply:
Sorry I didn't mean the AuthRequestValidator itself as that one is for validation against our /authorisation endpoint. I meant to create a new validator i.e TokenRequestValidator that can do the validation for the /token endpoint

Reply:
New class TokenRequestValidator added